CorsixTH/Lua/app.lua

```diff
@@ -1585,11 +1605,12 @@ function App:afterLoad()
                        ") to " .. new .. " (" .. self:getVersion() ..
                        "). The save was created using " .. first ..
                        " (" .. self:getVersion(first) .. ")")
-  else
-    -- TODO: This should maybe be forbidden completely.
+  else -- Save is newer than the game and can only proceed in debug mode
     self.world:gameLog("Warning: loaded savegame version " .. old .. " (" .. self:getVersion(old) ..
                        ")" .. " in older version " .. new .. " (" .. self:getVersion() .. ").")
+    self.ui:addWindow(UIInformation(self.ui, {_S.warnings.newersave}))
   end
+  self.world.release_version = self:getVersion()
   self.world.savegame_version = new
```

Comment:
Variables are created in world. Is that documented in World?

I am somewhat on a crusade to at least state in a constructor of a class which variables may exist, and what they mean. As the code is filled with this random creation of new variables outside the class, my journey may never end though :)